### ISSUE TITLE
bug: Regression: Tag value is optional, passing "" works on S3 but causes InternalError on localstack.

### ISSUE THREAD

**Issue by Greg-Myers-SB** [localstack/localstack#8598]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

With the following code:
```ruby
client = bucket.client
bucket.object_versions({ prefix: id }).each do |version|
  client.put_object_tagging(
    bucket: bucket.name,
    key: id,
    version_id: version.id,
    tagging: { tag_set: [{key: "my-tag", value: ""}] }
  )
end
```
On localstack 0.12 this applies the tag successfully as per [Tag Basics](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html#tag-basics) about value being optional.
On localstack 1.4.0 and 2.1.1.dev this fails with `Aws::S3::Errors::InternalError`

```
Aws::S3::Errors::InternalError:
       exception while calling s3.PutObjectTagging: Traceback (most recent call last):
         File "/opt/code/localstack/localstack/aws/chain.py", line 90, in handle
           handler(self, self.context, response)
         File "/opt/code/localstack/localstack/aws/handlers/service.py", line 123, in __call__
           handler(chain, context, response)
         File "/opt/code/localstack/localstack/aws/handlers/service.py", line 93, in __call__
           skeleton_response = self.skeleton.invoke(context)
         File "/opt/code/localstack/localstack/aws/skeleton.py", line 154, in invoke
           return self.dispatch_request(context, instance)
         File "/opt/code/localstack/localstack/aws/skeleton.py", line 166, in dispatch_request
           result = handler(context, instance) or {}
         File "/opt/code/localstack/localstack/aws/forwarder.py", line 60, in _call
           return handler(context, req)
         File "/opt/code/localstack/localstack/aws/skeleton.py", line 118, in __call__
           return self.fn(*args, **kwargs)
         File "/opt/code/localstack/localstack/aws/api/core.py", line 167, in operation_marker
           return fn(*args, **kwargs)
         File "/opt/code/localstack/localstack/services/s3/provider.py", line 731, in put_object_tagging
           response: PutObjectTaggingOutput = call_moto(context)
         File "/opt/code/localstack/localstack/services/moto.py", line 46, in call_moto
           return dispatch_to_backend(context, dispatch_to_moto, include_response_metadata)
         File "/opt/code/localstack/localstack/aws/forwarder.py", line 120, in dispatch_to_backend
           http_response = http_request_dispatcher(context)
         File "/opt/code/localstack/localstack/services/moto.py", line 111, in dispatch_to_moto
           response = dispatch(request, request.url, request.headers)
         File "/opt/code/localstack/localstack/utils/patch.py", line 38, in proxy
           return new(target, *args, **kwargs)
         File "/opt/code/localstack/localstack/services/s3/provider.py", line 1608, in _fix_key_response
           status_code, resp_headers, key_value = fn(self, *args, **kwargs)
         File "/opt/code/localstack/.venv/lib/python3.10/site-packages/moto/utilities/aws_headers.py", line 74, in _wrapper
           response = f(*args, **kwargs)
         File "/opt/code/localstack/.venv/lib/python3.10/site-packages/moto/s3/responses.py", line 1207, in key_response
           response = self._key_response(request, full_url, self.headers)
         File "/opt/code/localstack/.venv/lib/python3.10/site-packages/moto/s3/responses.py", line 1319, in _key_response
           return self._key_response_put(request, body, bucket_name, query, key_name)
         File "/opt/code/localstack/.venv/lib/python3.10/site-packages/moto/s3/responses.py", line 1614, in _key_response_put
           self.backend.set_key_tags(key_to_tag, tagging, key_name)
         File "/opt/code/localstack/.venv/lib/python3.10/site-packages/moto/s3/models.py", line 2037, in set_key_tags
           errmsg = self.tagger.validate_tags(boto_tags_dict)
         File "/opt/code/localstack/.venv/lib/python3.10/site-packages/moto/utilities/tagging_service.py", line 142, in validate_tags
           if len(tag_value) > 256:
       TypeError: object of type 'NoneType' has no len()
```

The rules according to S3:
```bash
$ aws --region my-region --profile my-profile s3api put-object-tagging --bucket my-bucket --key my-key --tagging '{"TagSet":[{"Key": "test"}]}'

Parameter validation failed:
Missing required parameter in Tagging.TagSet[0]: "Value"

$ aws --region my-region --profile my-profile s3api put-object-tagging --bucket my-bucket --key my-key --tagging '{"TagSet":[{"Key": "test", "Value": null}]}'

Parameter validation failed:
Invalid type for parameter Tagging.TagSet[0].Value, value: None, type: <class 'NoneType'>, valid types: <class 'str'>

$ aws --region my-region --profile my-profile s3api put-object-tagging --bucket my-bucket --key my-key --tagging '{"TagSet":[{"Key": "test", "Value": ""}]}'

$ aws --region my-region --profile my-profile s3api get-object-tagging --bucket my-bucket --key my-key
{
    "TagSet": [
        {
            "Key": "test",
            "Value": ""
        }
    ]
}
```

### Expected Behavior

On 1.4 and 2.x it should behave as it used to in 0.12.

Tag Value is optional when submitted as an empty string `""`

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

```docker-compose.yaml
version: '3.4'
services:
  localstack:
    image: localstack/localstack:${LOCALSTACK_VERSION:latest}
    ports:
      - "4566:4566"
    environment:
      - SERVICES=s3,kms,ssm
      - DEBUG=1
      - PERSISTENCE=/tmp/localstack/data
```

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

    $ awslocal s3 put-object-tagging s3://my-bucket --tagging '{"TagSet":[{"Key": "test", "Value": ""}]}' --key my-key

### Environment

```markdown
- OS: MacOS
- LocalStack: 2.1.1, 1.4.0, 0.12.11
```

**Comment by macnev2013** [comment]
Thanks for reporting the issue @Greg-Myers-SB. This has been fixed in #8638. Can you please try it once with the `latest` version of LocalStack? Please feel free to reach out if the issue persists.

Thank you!

**Comment by Greg-Myers-SB** [comment]
Thank you, that was a fast turnaround. Also in 0.12 it used to return "None" instead of "" for empty S3 tags. Now with the fix that you added, it correctly returns "" as the empty value. Well done on that also.
